artiq/applets/big_number.py

```diff
@@ -1,22 +1,68 @@
 #!/usr/bin/env python3
 
-from PyQt5 import QtWidgets
-
+from PyQt5 import QtWidgets, QtCore, QtGui
 from artiq.applets.simple import SimpleApplet
 
 
-class NumberWidget(QtWidgets.QLCDNumber):
+class QResponsiveLCDNumber(QtWidgets.QLCDNumber):
+    doubleClicked = QtCore.pyqtSignal()
+
+    def mouseDoubleClickEvent(self, event):
+        self.doubleClicked.emit()
+
+
+class NumberWidget(QtWidgets.QStackedWidget):
     def __init__(self, args, ctl):
-        QtWidgets.QLCDNumber.__init__(self)
-        self.setDigitCount(args.digit_count)
+        QtWidgets.QStackedWidget.__init__(self)
         self.dataset_name = args.dataset
+        self.ctl = ctl
+
+        self.lcd_widget = QResponsiveLCDNumber()
+        self.lcd_widget.setDigitCount(args.digit_count)
+        self.lcd_widget.doubleClicked.connect(self.edit_start)
+        self.addWidget(self.lcd_widget)
+
+        self.edit_widget = QtWidgets.QLineEdit()
+        self.edit_widget.setValidator(QtGui.QDoubleValidator())
+        self.edit_widget.editingFinished.connect(self.edit_finished)
+        self.addWidget(self.edit_widget)
+
+        font = QtGui.QFont()
+        font.setPointSize(60)
+        self.edit_widget.setFont(font)
+
+        self.setCurrentWidget(self.lcd_widget)
+        self.edit_cancelled = False
+
+    def keyPressEvent(self, event):
+        if event.key() == QtCore.Qt.Key_Escape:
+            self.edit_cancelled = True
+            self.setCurrentWidget(self.lcd_widget)
```

Comment:
I think adding it will not have any effect because the code only handles a single keyPress event. However if the widget (or its children) handle other key events, we should call the parent keyPressEvent.